File: Week 01/id_237/remove-duplicates-from-sorted-array.java

// 给定一个排序数组，你需要在原地删除重复出现的元素，使得每个元素只出现一次，返回移除后数组的新长度。

// 不要使用额外的数组空间，你必须在原地修改输入数组并在使用 O(1) 额外空间的条件下完成。

class Solution {
    public int removeDuplicates(int[] nums) {
        int m = 0;
        for(int i = 1; i < nums.length; i++){
            if(nums[i] != nums[m]){
                nums[++m] = nums[i];
            }
        }
        return m+1;
    }
}

File: Week 01/id_277/remove-duplicates-from-sorted-array.py
class Solution:
    """
    此题使用了双指针的解法;
    定义前指针和后指针,每次循环后指针的位置都向后移动一位;
    当发现后指针与前指针的数值不同时,再将后指针的数据复制到前指针的后一位,同时将前指针后移一位;
    """
    def removeDuplicates(self, nums: List[int]) -> int:
        before = 0
        after = 1

        while(after < len(nums)):
            if nums[before] != nums[after]:
                nums[before + 1] = nums[after]
                before +=1
            after +=1

        return before+1